### ISSUE TITLE
Whether create_graph=true or false, torch.autograd.grad returns different gradients.

### ISSUE THREAD

**Issue by vinnamkim** [pytorch/pytorch#41332]
## 🐛 Bug

Whether `create_graph=true` or `false`, `torch.autograd.grad` returns different gradients. Guess there are different flows on back-propagation of modules whether `create_graph=true` or `false`. What I found is only occurred on models in `transformer` package.

## To Reproduce

1. When using transformer model, the assertion of the following code is failed.

```
import torch
from transformers import BertForSequenceClassification, BertConfig

config = BertConfig()
model = BertForSequenceClassification(config=config).eval().cuda()

for k, v in model.named_parameters():
    v.data = 0.1 * torch.randn_like(v.data)
    
inputs_embeds = torch.randn([1, 100, config.hidden_size], requires_grad=True).cuda()
attention_mask = torch.ones([1, 100]).cuda()

outputs = model(inputs_embeds=inputs_embeds, attention_mask=attention_mask)[0]
grads1 = torch.autograd.grad(outputs[0, 1], inputs_embeds, create_graph=False)[0]

outputs = model(inputs_embeds=inputs_embeds, attention_mask=attention_mask)[0]
grads2 = torch.autograd.grad(outputs[0, 1], inputs_embeds, create_graph=True)[0]

print((grads1 - grads2).abs().max())
assert not ((grads1 - grads2).abs() > 1e-6).any()
```

Printing results
```
tensor(1.5196e+18, device='cuda:0', grad_fn=<MaxBackward1>)
```

2. In the other model(resnet50), there is no problem.

```
import torch
from torchvision.models import resnet50

model = resnet50().eval().cuda()
x = torch.randn([1, 3, 224, 224],requires_grad=True).cuda()

for k, v in model.named_parameters():
    v.data = 0.1 * torch.randn_like(v.data)
    
outputs = model(x)
grads1 = torch.autograd.grad(outputs[0, 1], x, create_graph=False)[0]

outputs = model(x)
grads2 = torch.autograd.grad(outputs[0, 1], x, create_graph=True)[0]

print((grads1 - grads2).abs().max())
assert not ((grads1 - grads2).abs() > 1e-6).any()
```

## Expected behavior

In reproduce step 1, the assertion should be succeeded.

## Environment

PyTorch version: 1.5.0+cu101
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
CMake version: version 3.13.3

Python version: 3.8
Is CUDA available: Yes
CUDA runtime version: 10.1.168
GPU models and configuration:
GPU 0: Tesla V100-SXM2-32GB
GPU 1: Tesla V100-SXM2-32GB
GPU 2: Tesla V100-SXM2-32GB
GPU 3: Tesla V100-SXM2-32GB
GPU 4: Tesla V100-SXM2-32GB
GPU 5: Tesla V100-SXM2-32GB
GPU 6: Tesla V100-SXM2-32GB
GPU 7: Tesla V100-SXM2-32GB

Nvidia driver version: 418.152.00
cuDNN version: /usr/local/cuda-10.1/targets/x86_64-linux/lib/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.18.1
[pip3] torch==1.5.1
[pip3] torch-geometric==1.6.0
[pip3] torch-scatter==2.0.4
[pip3] torch-sparse==0.6.1
[conda] blas                      1.0                         mkl
[conda] cudatoolkit               10.1.243             h6bb024c_0
[conda] mkl                       2020.0                      166
[conda] mkl-service               2.3.0            py38he904b0f_0
[conda] mkl_fft                   1.0.15           py38ha843d7b_0
[conda] mkl_random                1.1.0            py38h962f231_0
[conda] numpy                     1.18.2                   pypi_0    pypi
[conda] numpy-base                1.18.1           py38hde5b4d6_1
[conda] pytorch-nlp               0.5.0                     dev_0    <develop>
[conda] torch                     1.5.0+cu101              pypi_0    pypi
[conda] torchncmml                0.0.0                     dev_0    <develop>
[conda] torchvision               0.6.0+cu101              pypi_0    pypi

**Comment by amirgholami** [comment]
I have noticed the same problem and it would be great if someone could have a look

**Comment by yaozhewei** [comment]
If you downgraded the PyTorch to 1.3.1, you should resolve the issue. But a fix is definitely needed for the newer version.

**Comment by vinnamkim** [comment]
It is fixed. Close the issue.
